### PR TITLE
Publisher status refresh now shows when refresh is complete

### PR DESCRIPTION
**Pull request by jasonrsadler**
Fixes brave/brave-browser#4351
Fixes brave/brave-browser#4167
Fixes brave/brave-browser#4162

Brave UI PR: https://github.com/brave/brave-ui/pull/475

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on
  - [ ] Windows
  - [ ] macOS
  - [ ] Linux
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection

## Test Plan:

1. Start Brave with clean profile and enable Rewards.
2. Go to an unverified site, open panel and click the link to refresh publisher status.
3. Ensure a checkmark displays and fades after check is complete.
4. Quit browser
5. Modify `publisher_list` and remove a verified site.
6. Reopen browser and visit site that was removed from `publisher_list`
7. Open panel and verify publisher shows as unverified.
8. Click the link to refresh publisher status.
9. Ensure a checkmark displays and fades after check is complete, and also that publisher now shows as verified.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: components/brave_rewards/resources/extension/brave_rewards/_locales/en_US/messages.json [hunk 1]
```diff
@@ -298,7 +298,7 @@
     "description": "Error title when user missed a grant."
   },
   "unVerifiedCheck": {
-    "message": "Check Again...",
+    "message": "Refresh Status",
```
Comment (rossmoody) [inline components/brave_rewards/resources/extension/brave_rewards/_locales/en_US/messages.json]: Pls update line 51 to `Verified Creator` away from the term publisher.

Comment (jasonrsadler) [reply components/brave_rewards/resources/extension/brave_rewards/_locales/en_US/messages.json]: Done.

**Comment by rossmoody** [comment]
This Brave UI SHA update will inherit a few fixes from one previous PR that got merged so I added them to the description. Brave UI PR for reference: https://github.com/brave/brave-ui/pull/453

File: components/brave_rewards/resources/extension/brave_rewards/components/panel.tsx [hunk 2]
```diff
@@ -414,21 +416,41 @@ export class Panel extends React.Component<Props, State> {
...
+  initiateDelayCounter = () => {
+    setTimeout(() => {
+      this.setState({
+        timerPassed: true
+      })
+    }, 2000)
+  }
+
+  resetPublisherStatus = () => {
+    console.log(this.state.timerPassed)
```
Comment (ryanml) [inline components/brave_rewards/resources/extension/brave_rewards/components/panel.tsx]: log :3

File: components/brave_rewards/resources/extension/brave_rewards/components/panel.tsx [hunk 3]
```diff
@@ -414,21 +416,41 @@ export class Panel extends React.Component<Props, State> {
     return defaultContribution
   }
 
+  initiateDelayCounter = () => {
+    setTimeout(() => {
```
Comment (NejcZdovc) [inline components/brave_rewards/resources/extension/brave_rewards/components/panel.tsx]: this timeout needs to be cleared before set again. We also need to clear it on component unmount

File: components/brave_rewards/resources/extension/brave_rewards/components/panel.tsx [hunk 4]
```diff
@@ -414,21 +416,41 @@ export class Panel extends React.Component<Props, State> {
     return defaultContribution
   }
 
+  initiateDelayCounter = () => {
```
Comment (NejcZdovc) [inline components/brave_rewards/resources/extension/brave_rewards/components/panel.tsx]: is this delay fake? so it doesn't reflect actual response from c++?

What happens if connection is really slow and it takes 15s to get publisher list from the server and result from c++ into JS?

Comment (jasonrsadler) [reply components/brave_rewards/resources/extension/brave_rewards/components/panel.tsx]: Progress spinner is shown for minimum of 2 seconds. Can show for longer but this ensures that user at least sees the spinner.

File: components/brave_rewards/resources/extension/brave_rewards/components/panel.tsx (hunk 2 above)
Comment (jasonrsadler) [reply components/brave_rewards/resources/extension/brave_rewards/components/panel.tsx]: Removed.

File: components/brave_rewards/resources/extension/brave_rewards/components/panel.tsx [hunk 5]
```diff
@@ -414,21 +420,41 @@ export class Panel extends React.Component<Props, State> {
...
+    this.delayTimer = setTimeout(() => {
+      clearTimeout(this.delayTimer)
```
Comment (NejcZdovc) [inline components/brave_rewards/resources/extension/brave_rewards/components/panel.tsx]: this should be on the first line in `initiateDelayCounter`. Calling clear here doesn't do anything as when you are inside here timer will be automatically cleared after state is set

Comment (jasonrsadler) [reply components/brave_rewards/resources/extension/brave_rewards/components/panel.tsx]: Done.

**Comment by jasonrsadler** [comment]
timer check interval updated to 250ms

**Comment by alexwykoff** [comment]
Looks awesome! Thanks @jasonrsadler @NejcZdovc !

**Pull request merged**